app.R

```diff
@@ -114,17 +115,19 @@ render_trait_plot <- function(subexp_name, input, output, full_cache_data) {
     units <- selected_subexp_data[[ 'trait_data' ]][[ selected_variable ]][[ 'units' ]]
     title <- ifelse(units == '', selected_variable, paste0(selected_variable, ' (', units, ')'))
     
-    trait_plot <- ggplot() + 
-      geom_violin(data = plot_data, scale = 'width', width = 1,
-                   aes(x = as.Date(date), y = mean, 
-                       group = as.Date(date))) +
-      geom_boxplot(data = plot_data, 
-                  aes(x = as.Date(date), y = mean, 
-                      group = as.Date(date)), 
-                  outlier.alpha = 0.25, width = 0.2)  
-#      geom_point(data = plot_data, 
-#                 aes(x = as.Date(date), y = mean), 
-#                 alpha = 0.1, size = 0.1, position = position_jitter(width = 0.1))
+    num_unique_vals = length(unique(plot_data[[ 'mean' ]]))
+    
+    trait_plot <- ggplot(data = plot_data, aes(x = as.Date(date), y = mean))
+      {
+        if (num_unique_vals < 15) {
+          trait_plot <- trait_plot + geom_count() + geom_vline(aes(xintercept = as.numeric(as.Date(date))))
```

Comment:
Split each geom into a separate row

```r
trait_plot <- trait_plot + 
  geom_count() + 
  geom_vline(aes(xintercept = as.numeric(as.Date(date))))
```

Also try 

```
 geom_vline(aes(xintercept = as.numeric(as.Date(date))), color = 'darkgrey')
```

Reply:
Also try adding a legend when data is count data like:

```r
+ theme(legend.position = c(0.1, 0.9))
```

Reply:
ignore my suggested legend

* make vline grey or darkgrey 
* Set condition <= 20
* use <-

 if unique(plot_data[[mean]) == c(0,1) these are T/F instead of categorical. Ideas?